docs/aws/audit/elbmonitoring/rules/elb_nlb_listener_security_remediation.mdx

```diff
@@ -1,6 +1,252 @@
+
+4. **Configure Health Checks:**
+   - Select your NLB from the **Load Balancers** list.
+   - Go to the **Target Groups** tab and select the target group associated with your NLB.
+   - Under the **Health checks** tab, configure health checks to ensure that only healthy instances receive traffic. Use secure protocols for health checks if possible.
+
+By following these steps, you can ensure that your Network Load Balancer is configured securely in the AWS Management Console.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent Network Load Balancers (NLBs) from having insecure configurations in Elastic Load Balancer using AWS CLI, you can follow these steps:
+
+1. **Ensure SSL/TLS Termination:**
+   Make sure that your NLB is configured to use SSL/TLS termination to encrypt data in transit. This can be done by creating a listener with an SSL certificate.
+
+   ```sh
+   aws elbv2 create-listener --load-balancer-arn <load-balancer-arn> \
+   --protocol TLS --port 443 \
+   --certificates CertificateArn=<certificate-arn> \
+   --default-actions Type=forward,TargetGroupArn=<target-group-arn>
+   ```
+
+2. **Restrict Listener Ports:**
+   Ensure that only necessary ports are open on your NLB. For example, if you only need HTTPS, ensure that only port 443 is open.
+
+   ```sh
+   aws elbv2 delete-listener --listener-arn <listener-arn-for-port-80>
+   ```
+
+3. **Use Secure Security Groups:**
+   Attach a security group to your NLB that restricts inbound traffic to only necessary IP ranges and ports.
+
+   ```sh
+   aws ec2 authorize-security-group-ingress --group-id <security-group-id> \
+   --protocol tcp --port 443 --cidr <allowed-ip-range>
+   ```
+
+4. **Enable Access Logging:**
+   Enable access logging for your NLB to monitor and audit traffic, which can help in identifying and mitigating insecure configurations.
+
+   ```sh
+   aws elbv2 modify-load-balancer-attributes --load-balancer-arn <load-balancer-arn> \
+   --attributes Key=access_logs.s3.enabled,Value=true Key=access_logs.s3.bucket,Value=<s3-bucket-name> \
+   Key=access_logs.s3.prefix,Value=<s3-prefix>
+   ```
+
+By following these steps, you can ensure that your NLB is configured securely using the AWS CLI.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent Network Load Balancers (NLBs) from having insecure configurations in AWS Elastic Load Balancer using Python scripts, you can use the Boto3 library, which is the AWS SDK for Python. Here are the steps to ensure secure configurations:
+
+1. **Install Boto3 Library**:
+   Ensure you have the Boto3 library installed. If not, you can install it using pip:
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a Boto3 Session**:
+   Initialize a Boto3 session and create an ELB client.
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>